### DISCUSSION TITLE
Photoprism No stop crash :-/

### DISCUSSION THREAD

**Discussion by Ork7777** [photoprism/photoprism#4232] [Q&A]
Hi guys. I have a persistent problem with photoprism that has kept me stuck for months. I was thinking of using this program for my photo collection but for some reason I was never able to complete the scan . This is the second time I've reinstalled everything from scratch starting from docker desktop.

The problem is that during scanning photoprism (not the maria database) freezes or crashes. Sometimes after just a few minutes of operation, sometimes it lasts even ten minutes (¡!!)and then inevitably dies. Too bad that considering the size of my collection (+120.000 pictures) at this rate I would need about ten years to complete the process.
I'll give you the features of the system.
The server on which Docker Desktop runs is a windows 11 Ryzen 1700x with 16GB of RAM. The virtualization system used is wsl 2. I don't think I have memory problems since wsl 2 should handle it on its own from what I know. There is plenty of disk space available  Looking at the log it usually ends with the line: "/scripts/entrypoint.sh: line 118: 20 Killed"

Sifting through old posts here and on reddit I found similar situations and at first glance the culprit seems to be RAM or swap. But using wsl2 in theory this should be achieved automatically. To be safe I created "C:\Users\<UserName>\.wslconfig." with the following lines: "memory 12GB
swap 8GB
swapFile %USERPROFILE%\AppData\Local\Temp\swap.vhdx"
but nothing has changed. To tell the truth, I don't know if these changes have been taken into consideration, is there a way to verify that they are active?

If I wanted to try the preview version of photoprism what should I do? Changing the line: " photoprism:
    ## Use photoprism/photoprism:preview for testing preview builds:
    image: photoprism/photoprism:latest" in docker-compose.yml should be enough?

Any suggestions are welcome!
Thank you!

**Comment by Ork7777** [comment]
Ps. sometime there is no crash at all...
Photoprism starts to be unrensponsive...not dead but not alive yet... I can log in but the interface then is forever loading and there is nothing in the log.

![Screenshot 2024-05-04 131324](https://github.com/photoprism/photoprism/assets/168897439/4b0cc5c4-64a8-4a11-84e3-44f9ced3d1e7)

**Comment by alexanderhuzar** [comment]
Same issue here. Running on Docker Desktop and WSL is automatically managed. I even upgraded to 64 GB of RAM, but within minutes it crashes. I'm currently trying manually configuring the WSL2 memory usage limit as per this guide: https://emyode.com/en/blog/managing-resources-for-windows-subsystem-for-linux-wsl-on-a-windows-desktop/

TLDR:
Starting with Windows Insider build 18945, you can manually set the amount of memory WSL 2 is allowed to use. Add the following lines to your .wslconfig file located at C:\Users\<YourUsername>\.wslconfig:
[wsl2]
memory=48GB

So far no crashes, but it's only been about 10 minutes. Docker Desktop shows container memory usage is 1.7 GB at times which is insignificant. I have my fingers crossed and hope this is the only issue, the program itself seems rock solid otherwise and much better than IMMICH, but I wish I had the EXIF GPS coordinates in a mini-map on the right of every picture. Not a big deal as I'll take stability any day of the week.

Another thing that might possibly "fix" the issue, but I haven't tried it as the memory config above seems to have worked is:
PHOTOPRISM_WORKERS: 1
I tested this for about 1 minute, but it was under-utilizing my system and I have about 600,000 photos and videos that I'd like to index. Maybe this will work for those where the above does not work. Good luck!

**Reply by alexanderhuzar** [reply]
Yup, crashed 10 minutes after I wrote the above. I re-applied the "PHOTOPRISM_WORKERS: 1". I just tried with a value of 3, but it crashed. I want to see it run with 1 worker overnight, I might ramp it up to 2. This is so weird. The PhotoPrism directory is running off a Samsung 990 Pro 4TB SSD and the CPU is an i7-8700K so it's no slouch.

To reproduce: in Windows 11 Pro, install Docker Desktop, run in WSL2 mode. Install PhotoPrism onto your D: drive. Start caching hundreds of thousands of photos which are on D:/Photos.

**Reply by alexanderhuzar** [reply]
Nope, not stable with even 2 workers, going to have to revert to just 1 worker as that was good for since my last post above. I'm quite baffled why I need to cripple the performance like this just to not have my system BSOD. Wondering if it's some special instructions like heavy AVX or something that's causing this?

**Reply by alexanderhuzar** [reply]
I checked the WSL memory and swap space by going to CMD and typing in "wsl" which turns into WSL command prompt. There I typed in "free -h" and it showed me I have 64GB RAM with about 1.3 being used and swap space was 16 GB. I changed the .wslconfig file to say:
[wsl2]
memory=64GB
swap=32GB

I then confirmed this by typing in the "free -h" command after a reboot. Despite this - it had a BSOD after ~10 minutes of being busy at 95% total CPU usage across 6 cores (12 threads).

Trying to see if playing YouTube videos in Chrome is somehow related so keeping it closed for now, if stable for 1+ hour then I'll open it. Not sure how this is related as BSOD happens even with Chrome minimized and not playing anything, but it's a hunch.

**Reply by alexanderhuzar** [reply]
I officially give up. The only thing that seems to work is to set "PHOTOPRISM_WORKERS: 1". Nothing else works. It's running slowly, about 1/4 of what it could, but it's not causing BSOD. The machine this is running on is rock solid otherwise and was my former gaming system. Not overclocked, passes prime95, AVX tests like lindtpack, furmark, etc.

**Reply by lastzero** [reply]
I can't remember to have observed this in the past. Would it be possible that the WSL2 resource constraints have been added or changed only recently, for example with a Docker Desktop or Windows update? Note that  when using Hyper-V you always have to increase the memory and swap limit manually as described in our documentation. Very high  resolution images may also require more memory than mentioned in the system requirements.

**Reply by alexanderhuzar** [reply]
![image](https://github.com/user-attachments/assets/f9d7f827-d579-4143-b439-a77c0da699f5)
This isn't supposed to be very demanding, a couple of Chrome tabs probably use more RAM. I do have an older GTX 970 GPU, not sure if that another thing I can look at, but everything is running in software mode right now with no HW acceleration.

**Reply by alexanderhuzar** [reply]
Quick update: when it did BSOD, the sound output would get stuck in a rapid loop (really loud rapid stutter), the computer is locked up, then 10-20 seconds later the system would reboot. When checking Event Viewer I see this: 
![image](https://github.com/user-attachments/assets/68d72258-395f-405a-b8ea-161415accde5)
I don't use Intel Audio because I plug my TV into the HDMI port of my nVidia GPU.

Following advice from others online, I uninstalled from device manager the Intel Audio and clicked to attempt to remove the driver. Then I also disabled the Intel Audio service from Windows Services. Reboot.

I commented out the line "#PHOTOPRISM_WORKERS: 1" and so far so good, before it would almost immediately BSOD once the CPU usage spiked to 100%:
![image](https://github.com/user-attachments/assets/2e2e5f7c-07b1-463e-817b-69a886b682ee)

Why the eff is Intel Audio Service related in any way shape or form to this? It hasn't crashed with me doing anything else like Folding@Home, bitcoin mining, long gaming sessions, etc. Before PhotoPrism I had used IMMICH and it ran for weeks indexing and generating thumbnails and transcoding video and was 100% crash-free. Why would PhotoPrism on DockerDesktop cause Intel Audio Service to crash and then BSOD the system? I don't get it.

**Reply by lastzero** [reply]
From what I see, the problem occurs when converting RAW images with Darktable. So it could e.g. be caused by some system calls that Darktable uses, and not by PhotoPrism directly.

**Reply by alexanderhuzar** [reply]
I tried disabling Darktable - didn't work. I disabled all the bells and whistles like tensorflow as well. Even my 1 worker only also eventually crashes. I let it crash so many times, but it eventually indexed about 45,000 photos and videos. Problem is - I tried to create an album from one of my trips and share it. I viewed it in incognito and it wasn't populating the places (pictures on a map as per GPS coordinates). It was just pictures in folders. I tried scrolling down until I could see pictures with GPS coordinates hoping that once I see them they'd somehow be "re-cached" in the context of the album to make those appear on the places/states tab - but then I noticed my server crashed (again). I can't have this - my server is a web server, an *arr server, a NAS and performs many other functions. Again - not a single issue with IMMICH aside from the steep learning curve and no folder structure. PhotoPrism seems more efficient in what it does, but again - making my system BSOD is a deal-breaker.

I wanted PhotoPrism because of its supposed "zero bugs" policy, but this seems to have more bugs than Immich. It's a real shame too because I want to like PhotoPrism a lot more than Immich because of more features and especially access to folder structure. Immich just displays all pictures and does not present you with a folder structure as they're stored in the originals directory.

I'm unfortunately going to have to go back to Immich and manually create albums to correspond to the folder structure - I'd take that any day over random crashing. I can't accept that something else is causing the crashes because it only ever BSODs with PhotoPrism. The server can run indefinitely save for windows updates. This server has not once ever crashed since I got it in 2017. Not until PhotoPrism.

StableBit DrivePool issues with PhotoPrism: https://community.covecube.com/index.php?/topic/23730-drivepool-covefssys-bsodpc-reset-photoprism/

For any PhoPrism devs that might read this - I'd be happy to work with you and give you access to my system to test various things, inspect logs, whatever you need. In general the steps to reproduce are as follows:
On a Windows 11 Pro x64 machine - install Docker Desktop in WSL2 mode.
Install StableBit DrivePool and create a pool from any combination of drives (2 or more).
Place your photos (the more the better, but at minimum try 5,000 pictures and videos of all different types including RAW, HEIC, 8K, etc.) on the D: drive (or whatever the DrivePool directory is).
Install PhotoPrism onto the D: drive.
Configure your pictures: "D:/Pictures:/photoprism/originals/HTPC"
Leave all defaults in PhotoPrism including "insecure" password. Don't touch anything aside from maybe admin password.
Begin to index all originals.
Observe the (inevitable) BSOD.
Observe the Event Log fails to capture the cause of the BSOD.

I tried installing PhotoPrism onto a single SSD drive (Samsung 990 Pro 4TB) that isn't a StableBit-run volume, but it also crashes when the originals are stored in a DrivePool volume.

I also tried in addition to the above moving about 40K photos onto another SSD (root directory, not DrivePool volume), this works and confirms this: https://community.covecube.com/index.php?/topic/23730-drivepool-covefssys-bsodpc-reset-photoprism/ . I can't not have my photos in DrivePool. I can't not have DrivePool installed. And no I will not be running TrueNAS because I won't give up Windows just for that. I really really wanted to like and use PhotoPrism, but I'm done. The "it works on my computer" is not something that's going to work with me.

While I appreciate what docker containers can do in terms of being OS-agnostic, I don't plan on being in the middle of a blame game where PhoPrism blames DrivePool and DrivePool blames PhotoPrism. I don't have the patience for that. Sort it out yourselves. I can imagine how small of a team there is to be able to tackle this, but it is what it is. I'd love to give it a try again once this issue is nailed down and fixed, but not until then.

**Reply by alexanderhuzar** [reply]
Here's a screenshot showing IMMICH using vastly more resources and it works perfectly without BSODing my system: 
![2024-08-30 00_44_36-IMMICH hard at work no BSOD 2024-08-29 192119](https://github.com/user-attachments/assets/978edb06-791d-4391-91c4-b19d12a62a2a)


If IMMICH can do this (cause 42+GB RAM usage where it's just 11GB without Docker Desktop running) and read/write to/from a StableBit DrivePool volume then it's a simple process of elimination to figure out where to point the blame.
